start.sh

```diff
@@ -100,7 +100,18 @@ if [ -n "${PIHOLE_DNS_}" ]; then
     PIHOLE_DNS_ARR=(${PIHOLE_DNS_//;/ })
     count=1
     for i in "${PIHOLE_DNS_ARR[@]}"; do
-        change_setting "PIHOLE_DNS_$count" "$i"
+        # Split upstream address by #
+        split=(${i//#/ })
+
+        # Resolve hostname if any, otherwise use ip
+        resolved="$(ping -q -c 1 -t 1 "$split" | grep PING | sed -e "s/).*//" | sed -e "s/.*(//")"
```

Comment:
I like this idea, but I'm not sure this is the right way to go about it.

Could we use `dig +short "$split"`, instead? Less pipes is always a good thing!
pseudocode:

```
resolved="$(dig +short "$split")"

if (resolved is blank or empty)
    resolved=split 
fi

### continue as below
```